skyportal/handlers/api/analysis.py

```diff
@@ -0,0 +1,423 @@
+                        '`_authinfo` is required to add an Analysis Service '
+                        ' when authentication_type is not "none".'
+                    )
+                try:
+                    _authinfo = json.loads(_authinfo)
+                except json.JSONDecodeError:
+                    return self.error(
+                        '`_authinfo` must be parseable to a valid JSON object.'
+                    )
+                if authentication_type not in _authinfo:
+                    return self.error(
+                        f'`_authinfo` must contain a key for "{authentication_type}".'
+                    )
+
+        group_ids = data.pop('group_ids', None)
+        if not group_ids:
+            groups = self.current_user.accessible_groups
+        else:
+            try:
+                groups = Group.get_if_accessible_by(
+                    group_ids, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error('Could not find any accessible groups.', status=403)
+
+        schema = AnalysisService.__schema__()
+        try:
+            analysis_service = schema.load(data)
+        except ValidationError as e:
+            return self.error(
+                "Invalid/missing parameters: " f"{e.normalized_messages()}"
+            )
+
+        DBSession().add(analysis_service)
+        analysis_service.groups = groups
+
+        try:
+            self.verify_and_commit()
+        except IntegrityError as e:
+            return self.error(
+                f'Analysis Service with that name already exists: {str(e)}'
+            )
+
+        return self.success(data={"id": analysis_service.id})
+
+    @auth_or_token
+    def get(self, analysis_service_id=None):
+        """
+        ---
+        single:
+          description: Retrieve an Analysis Service by id
+          tags:
+            - analysis_services
+          parameters:
+            - in: path
+              name: analysis_service_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleAnalysisService
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all Analysis Services
+          tags:
+            - analysis_services
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfAnalysisServices
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+        if analysis_service_id is not None:
+            try:
+                s = AnalysisService.get_if_accessible_by(
+                    analysis_service_id, self.current_user, raise_if_none=True
```

Comment:
I think we usually now do raise_if_none=False and explicitly raise the error if None usually now. But your way seems fine